tests/keys.test.ts

```diff
@@ -10,175 +10,164 @@ beforeEach(async () => {
-    )
-    expect(defaultKey).toHaveProperty('key')
-    expect(defaultKey).toHaveProperty('actions')
-    expect(defaultKey).toHaveProperty('indexes')
-    expect(defaultKey).toHaveProperty('expiresAt', null)
-    expect(defaultKey).toHaveProperty('createdAt')
-    expect(defaultKey).toHaveProperty('updatedAt')
-
-    const adminKey = keys.find((key: Key) =>
-      key.description.startsWith('Default Admin API Key')
-    )
-
-    expect(adminKey).toBeDefined()
-    expect(adminKey).toHaveProperty(
-      'description',
-      'Default Admin API Key (Use it for all other operations. Caution! Do not use it on a public frontend)'
-    )
-    expect(adminKey).toHaveProperty('key')
-    expect(adminKey).toHaveProperty('actions')
-    expect(adminKey).toHaveProperty('indexes')
-    expect(adminKey).toHaveProperty('expiresAt', null)
-    expect(adminKey).toHaveProperty('createdAt')
-    expect(adminKey).toHaveProperty('updatedAt')
-  })
-
-  test(`${permission} key: get on key`, async () => {
-    const client = await getClient(permission)
-    const apiKey = await getKey('Private')
-
-    const key = await client.getKey(apiKey)
-
-    expect(key).toBeDefined()
-    expect(key).toHaveProperty(
-      'description',
-      'Default Admin API Key (Use it for all other operations. Caution! Do not use it on a public frontend)'
-    )
-    expect(key).toHaveProperty('key')
-    expect(key).toHaveProperty('actions')
-    expect(key).toHaveProperty('indexes')
-    expect(key).toHaveProperty('expiresAt', null)
-    expect(key).toHaveProperty('createdAt')
-    expect(key).toHaveProperty('updatedAt')
-  })
-
-  test(`${permission} key: create key with no expiresAt`, async () => {
-    const client = await getClient(permission)
-
-    const key = await client.createKey({
-      description: 'Indexing Products API key',
-      actions: ['documents.add'],
-      indexes: ['products'],
-      expiresAt: null,
-    })
+describe.each([{ permission: 'Master' }, { permission: 'Private' }])(
+  'Test on keys',
+  ({ permission }) => {
+    beforeEach(async () => {
+      const client = await getClient('Master')
+      await clearAllIndexes(config)
+
+      const keys = await client.getKeys()
 
-    expect(key).toBeDefined()
-    expect(key).toHaveProperty('description', 'Indexing Products API key')
-    expect(key).toHaveProperty('key')
-    expect(key).toHaveProperty('actions')
-    expect(key).toHaveProperty('indexes')
-    expect(key).toHaveProperty('expiresAt', null)
-    expect(key).toHaveProperty('createdAt')
-    expect(key).toHaveProperty('updatedAt')
-  })
-
-  test(`${permission} key: create key with an expiresAt`, async () => {
-    const client = await getClient(permission)
-
-    const key = await client.createKey({
-      description: 'Indexing Products API key',
-      actions: ['documents.add'],
-      indexes: ['products'],
-      expiresAt: '2050-11-13T00:00:00Z', // Test will fail in 2050
+      const customKeys = keys.results.filter(
```

Comment:
Why not just remove every key?

Reply:
Yes, this way you can ensure each of your tests will have what they need no matter what is the state of your Meilisearch instance. 

But since this could be led to a lot of changes I don't expect that for now, but could be in the future if you agree :)